core-build/gulp-core-build/src/tasks/JestTask.ts

```diff
@@ -152,7 +152,26 @@ export class JestTask extends GulpTask<IJestConfig> {
     const pattern: string = path.join(srcRoot, '**/__snapshots__/*.snap');
     globby.sync(pattern).forEach(sourceFile => {
       const destination: string = sourceFile.replace(srcRoot, destRoot);
-      fsx.copySync(sourceFile, destination);
+      const jsDestination: string = destination.replace(/\.test\..+\.snap$/, '.test.js.snap');
+      const jsxDestination: string = destination.replace(/\.test\..+\.snap$/, '.test.jsx.snap');
+      const tsDestination: string = destination.replace(/\.test\..+\.snap$/, '.test.ts.snap');
+      const tsxDestination: string = destination.replace(/\.test\..+\.snap$/, '.test.tsx.snap');
+
+      if (this._doesSnapBaseFileExist(jsDestination)) {
+        fsx.copySync(sourceFile, jsDestination);
+      } else if (this._doesSnapBaseFileExist(jsxDestination)) {
+        fsx.copySync(sourceFile, jsxDestination);
+      } else if (this._doesSnapBaseFileExist(tsDestination)) {
+        fsx.copySync(sourceFile, tsDestination);
+      } else if (this._doesSnapBaseFileExist(tsxDestination)) {
+        fsx.copySync(sourceFile, tsxDestination);
+      }
     });
   }
+
+  private _doesSnapBaseFileExist(snapFilePath: string): boolean {
+    const fileName: string = path.basename(snapFilePath, '.snap');
+    const snapBaseFilePath: string = path.resolve(snapFilePath, '../../', fileName);
```

Comment:
>'../../ [](start = 64, length = 7)

Is this always the same?

Reply:
This cannot be constructed from `destRoot` because that is only the **root**. The `../../` is used to go up from `__snapshots__` folder. I will put a comment here.